### DISCUSSION TITLE
Why show float value in meter? How to change to int?

### DISCUSSION THREAD

**Discussion by Krzych0409** [israel-dryer/ttkbootstrap#420] [Q&A] [answered]
I have default field 'stepsize' = 1. Why my meter allways show float value with '.0'? How to change it?

**Comment by Emooot** [comment]
I had the same issue. You will need to bind a function that will action when the amount used is changed.

def set_meter_value_int(*args):
    value = meterwidget["amountused"]
    # Set the value to the integer part of the current value
    meterwidget["amountused"]=(int(value))

meterwidget = Meter(mainframe, amountused=int(50)) # Or any default value
meterwidget.pack()
meterwidget("w", set_meter_value_int) # Will update the value to an integer when it is changed

EDIT: The last line is missing .amountusedvar.trace. See below for correct method.
meterwidget.amountusedvar.trace("w", set_meter_value_int) # Will update the value to an integer when it is changed

**Comment by Krzych0409** [comment]
Object is not method.

**Comment by Emooot** [comment]
Here it is in plain english:

1) Create a function that will retrieve the "amountused" value of the Meter and then set the meter "amountused" to an integer of that number.

2) Create the Meter and pack/grid it onto your GUI. 

3) Bind the function to the meter so that it is called whenever the value is changed.

**Reply by Krzych0409** [reply]
Problem is in 3 point. I have error: 
TypeError: 'Meter' object is not callable
meterwidget("w", set_meter_value_int)  <-- What is that?

**Comment by Emooot** [comment]
Ensure you import Meter from ttkbootstrap:

from ttkbootstrap import Meter

**Reply by Krzych0409** [reply]
LOL. How can you write in syntax: object()
It should be object.method()

**Reply by Emooot** [reply]
meterwidget.amountusedvar.trace("w", set_meter_value_int) # Will update the value to an integer when it is changed

**Reply by Krzych0409** [reply]
Now, I understand. Thanks

**Reply by Emooot** [reply]
No problem, hopefully you get it working.

**Reply by Krzych0409** [reply]
Yes, it is working. I do image resizer with meter.
[https://github.com/Krzych0409/image_resizer](https://github.com/Krzych0409/image_resizer)
I need to find a solution on how to remove the dot in place of the value. Do you have idea?

**Comment by israel-dryer** [comment, marked as answer]
Since v1.18.0 this is handled by the **`amountformat`** parameter — no trace/binding needed. It defaults to `"{:.0f}"`, so the meter already shows a plain integer (`50`, not `50.0`), and you can customize the display:

```python
from ttkbootstrap.widgets import Meter

Meter(app, amountused=50)                           # shows "50"
Meter(app, amountused=50, amountformat="{:.0f}%")   # shows "50%"
Meter(app, amountused=50, amountformat="{:.1f}")    # shows "50.0"
```

The earlier `amountusedvar.trace(...)` workaround still works but isn't necessary anymore (and it coerces the value itself to int). Use `amountformat` for display formatting instead.

Heads-up for **2.0**: this parameter is renamed `amount_format` (the old name still works with a deprecation warning), and the underlying value is float-backed so fractional formats like `"{:.1f}"` are fully honored.
